crates/optimism/chainspec/src/lib.rs

```diff
@@ -32,21 +32,53 @@ use reth_chainspec::{
     Hardforks, Head,
 };
 use reth_network_peers::NodeRecord;
+use reth_optimism_forks::OptimismHardfork;
 use reth_primitives_traits::Header;
 
+const DENOMINATOR_MASK: u64 = 0xFFFFFFFF00000000;
+const ELASTICITY_MASK: u64 = 0x00000000FFFFFFFF;
+
 /// OP stack chain spec type.
 #[derive(Debug, Clone, Deref, Into, Constructor, PartialEq, Eq)]
 pub struct OpChainSpec {
     /// [`ChainSpec`].
     pub inner: ChainSpec,
 }
 
+/// Fee trait for OP chain specs.
+pub trait Fee {
+    /// Read from parent to determine the base fee for the next block
+    fn next_block_base_fee(&self, parent: &Header, timestamp: u64) -> U256;
+}
+
 /// Returns the signature for the optimism deposit transactions, which don't include a
 /// signature.
 pub fn optimism_deposit_tx_signature() -> Signature {
     Signature::new(U256::ZERO, U256::ZERO, Parity::Parity(false))
 }
 
+impl Fee for OpChainSpec {
+    fn next_block_base_fee(&self, parent: &Header, timestamp: u64) -> U256 {
+        let is_holocene =
+            self.inner.is_fork_active_at_timestamp(OptimismHardfork::Holocene, timestamp);
+        if is_holocene {
```

Comment:
We need to also check if this is the _first_ holocene block by checking the parent timestamp. If it is, I believe the fee should use the same calculation as the else branch below.

Reference: https://specs.optimism.io/protocol/holocene/derivation.html#attributes-builder

See [kona's implementation](https://github.com/anton-rs/kona/blob/5cdb416c5f893b7bbb80cddf1570f724e69a6941/crates/derive/src/attributes/mod.rs#L209-L232).

Effectively, I believe we'll want to change the `if is_holocene {` line to `if is_holocene && not_first {` where
```rust
let not_first = self.inner.is_fork_active_at_timestamp(OptimismHardfork::Holocene, parent.timestamp);
```

Reply:
cc @clabby